xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -521,6 +523,23 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
         m_mime = "video/hevc";
         m_formatname = "amc-hevc";
       }
+
+      if (m_mime == "video/dolby-vision")
+      {
+        bool displaySupportsDovi =
+          CServiceBroker::GetWinSystem()->GetDisplayHDRCapabilities().SupportsDolbyVision();
+        bool mediaCodecSupportsDovi = CAndroidUtils::SupportsMediaCodecMimeType("video/dolby-vision");
+
+        CLog::Log(LOGDEBUG, "CDVDVideoCodecAndroidMediaCodec::Open Dolby Vision support: "
+                            "Display {}, MediaCodec: {}", displaySupportsDovi, mediaCodecSupportsDovi);
+
+        if (!displaySupportsDovi || !mediaCodecSupportsDovi)
```

Comment:
It seems to me as if the house had begun to be built from the roof and the order is illogical.

I mean: 

by default mime = video/hevc
If the system supports DV then mime = video/dolby-vision

something like this:

```c++
case AV_CODEC_ID_HEVC:
  
  m_mime = "video/hevc";
  m_formatname = "amc-hevc";

  const bool isDvhe = (m_hints.codec_tag == MKTAG('d', 'v', 'h', 'e'));
  const bool isDvh1 = (m_hints.codec_tag == MKTAG('d', 'v', 'h', '1'));

  if (isDvhe || isDvh1)
  {
    const bool displaySupportsDovi = CServiceBroker::GetWinSystem()->GetDisplayHDRCapabilities().SupportsDolbyVision();
    const bool mediaCodecSupportsDovi = CAndroidUtils::SupportsMediaCodecMimeType("video/dolby-vision");
      
    if (displaySupportsDovi && mediaCodecSupportsDovi)
    {
      if (isDvhe)
      {
        m_mime = "video/dolby-vision";
        m_formatname = "amc-dvhe";
      }
      else if (isDvh1)
      {
        m_mime = "video/dolby-vision";
        m_formatname = "amc-dvh1";
      }
    }
  }

. . . 

break;
```

Reply:
Thanks, that indeed does make a lot more sense.